modules/tinymce/src/core/main/ts/annotate/Identification.ts

```diff
@@ -42,19 +45,31 @@ const identify = (editor: Editor, annotationName: Optional<string>): Optional<{u
 
 const isAnnotation = (elem: any): boolean => SugarNode.isElement(elem) && Class.has(elem, Markings.annotation());
 
+const offscreenSelectionClass = 'mce-offscreen-selection';
+const hasOffscreenSelection = (body: SugarElement<Element>) =>
+  SelectorExists.descendant(body, `div.${offscreenSelectionClass}`);
+
+const isOffscreenSelectionElement = (elem: SugarElement<Node>) =>
+  Traverse.parent(elem).exists((parent) => Class.has(parent, offscreenSelectionClass));
+
 const findMarkers = (editor: Editor, uid: string): Array<SugarElement<Element>> => {
   const body = SugarElement.fromDom(editor.getBody());
-  return SelectorFilter.descendants(body, `[${Markings.dataAnnotationId()}="${uid}"]`);
+  const descendants = SelectorFilter.descendants(body, `[${Markings.dataAnnotationId()}="${uid}"]`);
+  const offscreenSelection = hasOffscreenSelection(body);
+  return offscreenSelection ? Arr.filter(descendants, (descendant) => !isOffscreenSelectionElement(descendant)) : descendants;
```

Comment:
This seems like a code smell to me... as this is one limited edge case and I think instead we need to be ignoring bogus elements in general not just the offscreen selection element.

- [ ] Swap to excluding `data-mce-bogus` elements instead (be careful of `all` vs `1`).

Reply:
I could be misunderstanding but were you thinking of something like this?

```
const isBogusElement = (elem: SugarElement<Element>, root: SugarElement<Node>) =>
  Attribute.has(elem, 'data-mce-bogus') || SelectorExists.ancestor(elem, '[data-mce-bogus="all"]', isRoot(root));

const findMarkers = (editor: Editor, uid: string): Array<SugarElement<Element>> => {
  const body = SugarElement.fromDom(editor.getBody());
  const descendants = SelectorFilter.descendants(body, `[${Markings.dataAnnotationId()}="${uid}"]`);
  return Arr.filter(descendants, (descendant) => !isBogusElement(descendant, body)) ;
};
```

Reply:
Yeah that's basically what I was thinking here.

Reply:
Thanks. I will make the relevant updates

Reply:
Have pushed up change. Thanks for the suggestion for a more general solution